### PR TITLE
Add Monte Carlo estimation of PI

### PR DESCRIPTION
**Pull request by cschuerc**
### **Describe your change:**

Added a Monte Carlo algorithm to estimate the mathematical constant PI. Doctests are not written since the algorithm outcome depends on RNG.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

File: maths/pi_monte_carlo_estimation.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        """
+        Generates a point randomly drawn from the unit square [0, 1) x [0, 1).
+        """
+        x = random.random()
+        y = random.random()
+
+        return cls(x, y)
+
+
+def estimate_pi(number_of_simulations):
```
[the remark this answers is not in the document]
Comment (cschuerc) [reply maths/pi_monte_carlo_estimation.py]: Thanks. I changed this. I wanted to change:
`def random_unit_square(cls):`
as well to
`def random_unit_square(cls) -> Point:`
but Python throws
_NameError: name 'Point' is not defined_
so I omitted this.

**Comment by cclauss** [comment]
What am I doing wrong?
```python
from math import pi
for number_of_simulations in range(1, 5000):
    if abs(estimate_pi(number_of_simulations) - pi) < 0.000:
        print("It took {number_of_simulations} Monte Carlo simulations to "
              "estimate pi to within 3 decimal places.")
```

File: maths/pi_monte_carlo_estimation.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    for simulation_index in range(number_of_simulations):
+        random_point = Point.random_unit_square()
+
+        if random_point.is_in_unit_circle():
+            number_in_unit_circle += 1
+
+    return 4 * number_in_unit_circle / number_of_simulations
+
+
+if __name__ == "__main__":
```
Comment (cclauss) [inline maths/pi_monte_carlo_estimation.py]: Let's show the accuracy of our estimation vs. the Python standard library's `math.py`.
```
if __name__ == "__main__":
    # import doctest

    # doctest.testmod()
    from math import pi
    prompt = "Please enter the desired number of Monte Carlo simulations: "
    number_of_simulations = int(input(prompt).strip())
    my_pi = estimate_pi(number_of_simulations)
    print(f"An estimate of PI is {my_pi} with an accuracy of {abs(my_pi - pi)}")
```

**Comment by cschuerc** [comment]
> 
> 
> What am I doing wrong?
> 
> ```python
> from math import pi
> for number_of_simulations in range(1, 5000):
>     if abs(estimate_pi(number_of_simulations) - pi) < 0.000:
>         print(f"It took {number_of_simulations} Monte Carlo simulations to "
>                "estimate pi to within 3 decimal places.")
> ```

The error can't be < 0.000, probably you wanted to write < 0.001? However, the estimates of PI get more exact with higher `number_of_simulations` but only in a statistical sense. So it could be that for `number_of_simulations == 10` the error is < 0.001 and for `number_of_simulations == 1000` it is not.

File: maths/pi_monte_carlo_estimation.py (hunk 2 above)
Comment (cschuerc) [reply maths/pi_monte_carlo_estimation.py]: Good point. I added it.

**Comment by cschuerc** [comment]
@cclauss How should we proceed with this?

File: maths/pi_monte_carlo_estimation.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    def __init__(self, x: float, y: float) -> None:
+        self.x = x
+        self.y = y
+
+    def is_in_unit_circle(self) -> bool:
+        """
+        True, if the point lies in the unit circle
+        False, otherwise
+        """
+        return (self.x ** 2 + self.y ** 2) <= 1
```
Comment (TheSuperNoob) [inline maths/pi_monte_carlo_estimation.py]: As a side note; `return self.x * self.x + self.y * self.y <= 1` is probably going to be faster and more efficient than using pythons power notation.

File: maths/pi_monte_carlo_estimation.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @classmethod
+    def random_unit_square(cls):
+        """
+        Generates a point randomly drawn from the unit square [0, 1) x [0, 1).
+        """
+        x = random.random()
+        y = random.random()
+
+        return cls(x, y)
```
Comment (cclauss) [inline maths/pi_monte_carlo_estimation.py]:
```suggestion
        return cls(x = random.random(), y = random.random())
```
and lose lines 21, 22, 23.  Then I think we can land this one.

**Pull request merged**